crates/phala-trie-storage/src/memdb.rs

```diff
@@ -237,38 +293,25 @@ where
     /// when the refs > 0.
     pub fn raw(&self, key: &<H as KeyHasher>::Out, prefix: Prefix) -> Option<(&T, i32)> {
         if key == &self.hashed_null_node {
-            return Some((&self.null_node_data, 1));
+            return Some((&self.null_node_data, 1))
         }
-        self.data
-            .get(&KF::key(key, prefix))
-            .map(|(value, count)| (value, *count))
+        self.data.get(&KF::key(key, prefix)).map(|(value, count)| (value, *count))
     }
 
     /// Consolidate all the entries of `other` into `self`.
     pub fn consolidate(&mut self, mut other: Self) {
         for (key, (value, rc)) in other.drain() {
-            if rc == 0 {
-                continue;
-            }
             match self.data.entry(key) {
                 Entry::Occupied(mut entry) => {
                     if entry.get().1 < 0 {
-                        self.malloc_tracker.on_insert(&value);
-                        self.malloc_tracker.on_remove(&entry.get().0);
                         entry.get_mut().0 = value;
                     }
 
                     entry.get_mut().1 += rc;
-
-                    if entry.get().1 == 0 {
-                        let (value, _) = entry.remove();
```

Comment:
ditto, we remove the the data imediately rather than in a later purge call. This can remove the `GC-stop-the-world` effect while syncing block into pruntime in SGX.